nautobot/extras/api/views.py

```diff
@@ -599,11 +599,19 @@ def run(self, request, *args, **kwargs):
                 request.method, detail="This job's source code could not be located and cannot be run"
             )
 
+        valid_queues = [settings.CELERY_TASK_DEFAULT_QUEUE]
         job_queues = job_model.job_queues.all().values_list("name", flat=True)
-        valid_queues = job_queues if job_queues else [settings.CELERY_TASK_DEFAULT_QUEUE]
+        if job_queues:
+            valid_queues += list(job_queues)
         # Get a default queue from either the job model's specified task queue or
         # the system default to fall back on if request doesn't provide one
-        default_valid_queue = valid_queues[0]
+        if job_model.default_job_queue:
+            default_valid_queue = job_model.default_job_queue.name
+        else:
+            default_valid_queue, _ = JobQueue.objects.get_or_create(
+                name=settings.CELERY_TASK_DEFAULT_QUEUE, defaults={"queue_type": JobQueueTypeChoices.TYPE_CELERY}
+            )
+            default_valid_queue = default_valid_queue.name
```

Comment:
I don't think we should be creating a new JobQueue as a side effect of calling the API. If a job has no default queue, and CELERY_TASK_DEFAULT_QUEUE doesn't exist, and the user didn't specify a queue, it feels to me like we should reject the request.